### ISSUE TITLE
Get-WillowInstalledProducts fails when package cache has been moved from the default location

### ISSUE THREAD

**Issue by flcdrg** [jberezanski/ChocolateyPackages#49]
Visual Studio 2017 supports moving the package cache from the default location via setting registry keys. See https://blogs.msdn.microsoft.com/heaths/2017/04/17/moving-or-disabling-the-package-cache-for-visual-studio-2017/ for more information.

`Get-WillowInstalledProducts` does support passing in a BasePath parameter, but from what I can see it always uses the default value.

I think this script should check those registry locations if a BasePath value is not provided before giving up on finding 2017

**Comment by flcdrg** [comment]
Something like this

```powershell
# Package cache may have been moved, so check registry - https://blogs.msdn.microsoft.com/heaths/2017/04/17/moving-or-disabling-the-package-cache-for-visual-studio-2017/
$cachePath = (Test-RegistryValue HKLM:\SOFTWARE\Policies\Microsoft\VisualStudio\Setup -Name CachePath -PassThru),
  (Test-RegistryValue HKLM:\SOFTWARE\Microsoft\VisualStudio\Setup -Name CachePath -PassThru),
  (Test-RegistryValue HKLM:\SOFTWARE\WOW6432Node\Microsoft\VisualStudio\Setup -Name CachePath -PassThru) | Select-Object -First 1

$basePath = if ($cachePath) { [IO.Path]::Combine($cachePath.CachePath, "_Instances") } else { "$Env:ProgramData\Microsoft\VisualStudio\Packages\_Instances" }
```

**Comment by bryan5989** [comment]
I just got bit by this limitation, so I submitted the PR that @jberezanski requested.

I realize this is a year old issue, but It's still affecting some.

**Comment by jberezanski** [comment]
The fix has just been published in version 1.8.1, together with #9.
